Simplified/NYPLBarcodeScanningViewController.m

```diff
@@ -0,0 +1,264 @@
+		scaleVideoX = self.view.frame.size.width / videoSizeX;
+		scaleVideoY = self.view.frame.size.height / videoSizeY;
+		scaleVideo = MAX(scaleVideoX, scaleVideoY);
+		if(scaleVideoX > scaleVideoY) {
+			transformedVideoRect.origin.y += (scaleVideo * videoSizeY - self.view.frame.size.height) / 2;
+		} else {
+			transformedVideoRect.origin.x += (scaleVideo * videoSizeX - self.view.frame.size.width) / 2;
+		}
+	} else {
+		scaleVideoX = self.view.frame.size.width / videoSizeY;
+		scaleVideoY = self.view.frame.size.height / videoSizeX;
+		scaleVideo = MAX(scaleVideoX, scaleVideoY);
+		if(scaleVideoX > scaleVideoY) {
+			transformedVideoRect.origin.y += (scaleVideo * videoSizeX - self.view.frame.size.height) / 2;
+		} else {
+			transformedVideoRect.origin.x += (scaleVideo * videoSizeY - self.view.frame.size.width) / 2;
+		}
+	}
+	_captureSizeTransform = CGAffineTransformMakeScale(1/scaleVideo, 1/scaleVideo);
+	self.capture.scanRect = CGRectApplyAffineTransform(transformedVideoRect, _captureSizeTransform);
+}
+
+#pragma mark - Private Methods
+
+- (NSString *)barcodeFormatToString:(ZXBarcodeFormat)format {
+  switch (format) {
+    case kBarcodeFormatAztec:
+      return @"Aztec";
+
+    case kBarcodeFormatCodabar:
+      return @"CODABAR";
+
+    case kBarcodeFormatCode39:
+      return @"Code 39";
+
+    case kBarcodeFormatCode93:
+      return @"Code 93";
+
+    case kBarcodeFormatCode128:
+      return @"Code 128";
+
+    case kBarcodeFormatDataMatrix:
+      return @"Data Matrix";
+
+    case kBarcodeFormatEan8:
+      return @"EAN-8";
+
+    case kBarcodeFormatEan13:
+      return @"EAN-13";
+
+    case kBarcodeFormatITF:
+      return @"ITF";
+
+    case kBarcodeFormatPDF417:
+      return @"PDF417";
+
+    case kBarcodeFormatQRCode:
+      return @"QR Code";
+
+    case kBarcodeFormatRSS14:
+      return @"RSS 14";
+
+    case kBarcodeFormatRSSExpanded:
+      return @"RSS Expanded";
+
+    case kBarcodeFormatUPCA:
+      return @"UPCA";
+
+    case kBarcodeFormatUPCE:
+      return @"UPCE";
+
+    case kBarcodeFormatUPCEANExtension:
+      return @"UPC/EAN extension";
+
+    default:
+      return @"Unknown";
+  }
+}
+
+#pragma mark - ZXCaptureDelegate Methods
+
+- (void)captureResult:(ZXCapture *)__unused capture result:(ZXResult *)result {
+  if (!result) return;
+
+	CGAffineTransform inverse = CGAffineTransformInvert(_captureSizeTransform);
+	NSMutableArray *points = [[NSMutableArray alloc] init];
+	NSString *location = @"";
+	for (ZXResultPoint *resultPoint in result.resultPoints) {
+		CGPoint cgPoint = CGPointMake(resultPoint.x, resultPoint.y);
+		CGPoint transformedPoint = CGPointApplyAffineTransform(cgPoint, inverse);
+		transformedPoint = [self.view convertPoint:transformedPoint toView:self.view.window];
+		NSValue* windowPointValue = [NSValue valueWithCGPoint:transformedPoint];
+		location = [NSString stringWithFormat:@"%@ (%f, %f)", location, transformedPoint.x, transformedPoint.y];
+		[points addObject:windowPointValue];
```

Comment:
does points get used?  is there ever a time when a barcode might not be recognized, even though the points are, like maybe when a barcode is smudged?

Reply:
Refer to above comment.